plugin/src/main/java/org/gradle/testretry/internal/executer/framework/Junit5TestFrameworkStrategy.java

```diff
@@ -30,9 +30,15 @@
 
 final class Junit5TestFrameworkStrategy extends BaseJunitTestFrameworkStrategy {
 
+    private final boolean isSpock2Used;
+
+    public Junit5TestFrameworkStrategy(boolean isSpock2Used) {
+        this.isSpock2Used = isSpock2Used;
+    }
+
     @Override
     public TestFramework createRetrying(TestFrameworkTemplate template, TestFramework testFramework, TestNames failedTests) {
-        DefaultTestFilter failedTestsFilter = testFilterFor(failedTests, false, template);
+        DefaultTestFilter failedTestsFilter = testFilterFor(failedTests, isSpock2Used, template);
```

Comment:
💬 This boolean will be propagated as`canRunParameterizedSpockMethods` to https://github.com/gradle/test-retry-gradle-plugin/blob/main/plugin/src/main/java/org/gradle/testretry/internal/executer/framework/BaseJunitTestFrameworkStrategy.java#L91 that ultimately decides how to retry parameterized test failures: as a whole class or individually.